sympy/ntheory/residue_ntheory.py

```diff
@@ -615,16 +615,59 @@ def is_nthpow_residue(a, n, m):
 
     P. Hackman "Elementary Number Theory" (2009),  page 76
     """
-    if( primitive_root(m) == None ):
-        raise NotImplementedError("Not implemented for m which do not have any primitive root")
-    if n == 1:
-        return True
     if n == 2:
         return is_quad_residue(a, m)
+    if n == 1:
+        return True
+    if primitive_root(m) == None:
+        x = factorint(m)
+        for prime, power in x.iteritems():
```

Comment:
This could probably be ``x.items()``. ``iteritems`` does not seem to be supported in Python 3. (https://docs.python.org/3/whatsnew/3.0.html?highlight=iteritems#views-and-iterators-instead-of-lists)